controllers/users.js

```diff
@@ -1,37 +1,125 @@
-const jwt = require("jsonwebtoken");
+const jwt = require('jsonwebtoken');
 
-function userDefinedException(message, statusCode) {
-  this.message = message;
-  this.statusCode = statusCode;
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
 }
+
+const MAX_RETRY_COUNT = 4;
+
+function isStoredTimeWithin10Minutes(otpGeneratedAt) {
+  // Convert stored time to Date object
+  const storedDate = new Date(otpGeneratedAt);
+
+  // Get current time
+  const currentDate = new Date();
+
+  // Calculate difference in milliseconds
+  const differenceInMilliseconds = currentDate - storedDate;
+
+  // Convert difference to minutes
+  const differenceInMinutes = differenceInMilliseconds / (1000 * 60); // 1000 milliseconds in a second, 60 seconds in a minute
+
+  // Check if difference is less than 10 minutes
+  return differenceInMinutes < 10;
+}
+
 function auth(app, sequelize) {
   const { User } = sequelize;
-  app.post("/login", async function login(req, res) {
-    const { username, password } = req.body;
+  app.post('/sendOtp', async function sendOtp(req, res) {
+    const { mobileNumber } = req.body;
     try {
-      if (!username || !password) {
-        throw new userDefinedException(
-          "Please enter both Email and Password",
-          401
-        );
+      if (!mobileNumber) {
+        return res.status(401).json({ message: 'Please provide with mobile number!' });
       }
-      const user = await User.findOne({ where: { username } });
-      console.log(user);
+      const user = await User.findOne({ where: { mobileNumber } });
       if (!user) {
-        throw new userDefinedException("No user found!", 404);
+        return res.status(404).json({ message: 'User not found!' });
       }
-      if (password !== user.password) {
-        throw new userDefinedException("Wrong password!", 401);
+
+      if (!user.otp) {
+        const randomInt = getRandomInt(1000, 9999); // Generates a random integer between 1000 and 9999
+        const otpGeneratedAt = new Date();
+        let retryCount = user.retryCount + 1;
+        //add send Interakt notification function
+        const updatedUser = await user.update({
+          otp: randomInt,
+          retryCount,
+          otpGeneratedAt
+        });
+        console.log(updatedUser);
+        return res.status(200).json({ message: `Otp sent successfully to ${mobileNumber}` });
       }
-      const token = jwt.sign(
-        { username: user.username },
-        process.env.JWT_SECRET,
-        {
-          expiresIn: "24h",
+      const isOtpValid = isStoredTimeWithin10Minutes(user.otpGeneratedAt);
+      if (user.retryCount < MAX_RETRY_COUNT && isOtpValid) {
```

Comment:
I'd reverse the logic, start with eliminating the conditions.

```suggestion
      if (!isOtpValid) {
       // do something 
      }
      
      else if(user.retryCount > MAX_RETRY_COUNT){
       // do stuff here
      }
      
      // send successful response. 
```

This will make the code easier to reason about.

Reply:
got it, fixed